micropip/_micropip.py

```diff
@@ -89,6 +89,12 @@ def from_url(url: str) -> "WheelInfo":
             parsed_url=parsed_url,
         )
 
+    def best_compatible_tag_index(self) -> int | None:
+        for index, tag in enumerate(sys_tags()):
+            if tag in self.tags:
+                return index
+        return None
+
     def is_compatible(self):
         if self.filename.endswith("py3-none-any.whl"):
             return True
```

Comment:
Maybe replace the section after this by,
```
return self.best_compatible_tag_index() is not None
```
to avoid having similar duplicate code paths.